backend/src/api/cycle.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const { Cycle } = require('../models')
+const { directorsOnly, errorWrap } = require('../middleware')
+const router = express.Router()
+
+// Get all cycles (previous and current)
+router.get(
+  '/',
+  errorWrap(async (req, res) => {
+    const cycles = await Cycle.find()
+    res.json({
+      code: 200,
+      result: cycles,
+      success: true
+    })
+  })
+)
+
+// Get a cycle by ID
+router.get(
+  '/:cycle_id',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const cycleId = req.params.cycle_id
+    const cycle = await Cycle.findById(cycleId)
+    res.json({
+      code: 200,
+      result: cycle,
+      success: true
+    })
+  })
+)
+
+// Create a new cycle
+router.post(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    let response = 'Cycle Created Successfully'
+    let code = 404
+    // TODO: not sure if this should be 404 here
+
+    const newTerm = req.body.term
+    const newChapter = req.body.chapter
+
+    if (!newTerm) {
+      response = 'Invalid term'
+    }
+    if (!newChapter) {
+      response = 'Invalid chapter'
+    }
+
+    const cycle = new Cycle({
+      term: newTerm,
+      chapter: newChapter
+    })
```

Comment:
This cycle is created even if the term or chapter is invalid. Maybe consider using `else if` and `else` to check if chapter is invalid and create the cycle respectively. You could also leave it as is and just rely on `cycle.save()` throwing an error, but that won't happen right now either since you haven't required those fields.

Reply:
Thanks for the feedback!! so as I understand, if those fields are required then this part will be fine?

Reply:
I'm not sure that this would be fine (think the await cycle.save() will throw an error but the message you want won't be surfaced correctly), but we can text/fix in a follow up revision